### PR TITLE
handled ad hoc multireddit request_promo bug

### PR DESCRIPTION
**Pull request by jamesmaa**
JIRA Ticket:
https://reddit.atlassian.net/browse/ADS-424

Regex Debugger: 
https://www.debuggex.com/

### CODE REVIEW

**Comment by jamesmaa** [comment]
👓 : @dwick @zeantsoi

File: reddit_adzerk/lib/validator.py [hunk 1]
```diff
@@ -8,9 +8,11 @@
 from r2.models import (
     NotFound,
     Subreddit,
+    MultiReddit,
 )
 
 is_multi_rx = re.compile(r"\A/?(user|r)/[^\/]+/m/(?P<name>.*?)/?\Z")
+is_adhoc_multi_rx = re.compile(r"\A\/r\/[^\/]+([0-z]+)(\+[0-z]+)+\Z")
```
Comment (zeantsoi) [inline reddit_adzerk/lib/validator.py]: This doesn't quite seem to do what I think we're trying to accomplish.

http://pythex.org/?regex=\A\%2Fr\%2F[^\%2F]%2B%28[0-z]%2B%29%28\%2B[0-z]%2B%29%2B\Z&test_string=%2Fr%2Fpics%2BEarthPorn%2Bfunny&ignorecase=0&multiline=0&dotall=0&verbose=0

Comment (jamesmaa) [inline reddit_adzerk/lib/validator.py]: i thought
/r/pics+EarthPorn+funny
is supposed to be a valid string, or am i mistaken?

Comment (dwick) [inline reddit_adzerk/lib/validator.py]: it is, z can you be more specific?

**Comment by dwick** [comment]
🐟 this lgtm

File: reddit_adzerk/lib/validator.py [hunk 2]
```diff
@@ -23,6 +25,10 @@ def run(self, path):
 
         if is_multi_rx.match(path):
             return VMultiByPath(self.param, kinds=("m")).run(path)
+        elif is_adhoc_multi_rx.match(path):
+            sr_strings = path.replace("/r/","").split("+")
+            srs = Subreddit._by_name(sr_strings).values()
```
Comment (bsimpson63) [inline reddit_adzerk/lib/validator.py]: please add `stale=True`

**Comment by bsimpson63** [comment]
Can you update your commit message to follow https://github.com/reddit/styleguide/tree/master/general-style-guidelines#commits

File: reddit_adzerk/lib/validator.py (hunk 1 above)
Comment (aoiwelle) [inline reddit_adzerk/lib/validator.py]: I think Z is commenting(correct me if I'm wrong) that you're not quite capturing what you think you are. That said, you could get away with capturing the substring and split on that down below. I would try
```python
is_adhoc_multi_rx = re.compile("\A\/r\/((?:[0-z]+\+)+(?:[0-z])+)\Z")
```
then below, you could
```python
adhoc_mutli_rxes = is_adhoc_multi_rx.match(path)
...
elif adhoc_multi_rxes:
    sr_strings = adhoc_multi_rxes.group(1).split("+")
...
```

Comment (aoiwelle) [inline reddit_adzerk/lib/validator.py]: [The Regex](http://pythex.org/?regex=%5CA%5C%2Fr%5C%2F((%3F%3A%5B0-z%5D%2B%5C%2B)%2B(%3F%3A%5B0-z%5D)%2B)%5CZ&test_string=%2Fr%2Fpics%2BEarthPorn%2Bfunny&ignorecase=0&multiline=0&dotall=0&verbose=0) yields matching only on  the desired substring `"pics+EarthPorn+funny"`

Comment (jamesmaa) [inline reddit_adzerk/lib/validator.py]: Thanks Neal! #regexmaster

Comment (zeantsoi) [inline reddit_adzerk/lib/validator.py]: Yup, that's what I was getting at. Neal's suggestion makes sense to me, thanks!

**Comment by zeantsoi** [comment]
Just [Brian's comment](https://github.com/reddit/reddit-plugin-adzerk/pull/181#issuecomment-216270645), else :fish:

**Pull request merged**